client/src/components/ColorList.js

```diff
@@ -18,15 +23,43 @@ const ColorList = ({ colors, updateColors }) => {
 
   const saveEdit = e => {
     e.preventDefault();
-    // Make a put request to save your updated color
-    // think about where will you get the id from...
-    // where is is saved right now?
+    updateColor(colorToEdit)
   };
 
+  const handleNewColor = e => {
+    e.preventDefault();
+    addColor();
+  }
+
+  const updateColor = () => {
+    axiosWithAuth()
+      .put(`/api/colors/${colorToEdit.id}`, colorToEdit)
+      .then(() => {
+        fetchColor()
+      })
+      .catch(error => console.log(error))
+      setColorToEdit({color: "", code: { hex: "" }})
+  }
+
   const deleteColor = color => {
-    // make a delete request to delete this color
+    axiosWithAuth()
+      .delete(`/api/colors/${color.id}`, colorToEdit)
+      .then(() => {
+        fetchColor()
+      })
+      .catch(error => console.log(error))
   };
 
+  const addColor = () => {
+    axiosWithAuth()
+      .post(`/api/colors`, newColor)
+      .then(response => {
+        console.log(response)
+        fetchColor()
+      })
+      .catch(error => console.log(error))
+  }
+
```

Comment:
For both adding and editing, I'd also add extra logic to make sure the forms disappear if the post/put is successful '' '